### ISSUE TITLE
Pasting (ctrl+v) often results in a spinning circle (appears connected to COPILOT).

### ISSUE THREAD

**Issue by Jawmo** [microsoft/vscode#236826]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.96.2
- OS Version: Windows 10 22H2

Steps to Reproduce:

1. Ctrl+c or ctrl+x some text.
2. Ctrl-v somewhere else in the same file.

This just started happening around 2 weeks ago. I started getting a spinning/loading circle everytime I try to paste code. It doesn't matter how much text, it can even be a single letter. It appears to be the same loading indicator that I see when copilot is thinking and, yes, I do have copilot enabled and am using it, but have had it for at least 6 months and this problem just started happening a few weeks ago.

It does seem to go away when I disable completions on copilot and returns when I re-enable them in limited testing.

I just saw this other bug here [https://github.com/microsoft/vscode/issues/193610 but I](https://github.com/microsoft/vscode/issues/193610) but it has no solution within it and I could not add to it.

Glad to provide more info if needed.

**Comment by vastava** [comment]
This has been happening to me too. For me, the spinning wheel appears when I paste content from within the same file or another file in VSCode. However, copying and pasting text from an external source (eg. notes app) doesn’t trigger the loading symbol.

**Comment by kadr2022** [comment]
Happening to me as well. I have enabled Copilot. It started happening since Copilot extension is enabled.

**Comment by d3lay** [comment]
I'm having the same error on Windows and I do not have the Copilot extension installed

**Comment by musme** [comment]
> I'm having the same error on Windows and I do not have the Copilot extension installed

I have the same issue

**Comment by joaomoreno** [comment]
Could this be a dupe of #236554?

**Comment by Jawmo** [comment]
Feels to me like copilot is trying to perform a completion or suggestion when I am pasting something before I give it a moment to complete that suggestion. If I wait until after it provides something, then it pastes normally. Seems like copilot is interrupting my ability to paste until it completes its thought. 

Definitely a huge pain in the butt in terms of slowing down my workflow.

**Comment by Sunrisies** [comment]
这个我也遇到了，有什么解决方法吗？

**Comment by skykep** [comment]
Experiencing the same thing.  If I try to copy text from another open file, it will hang and say "Analyzing 'sql.js' and its dependencies" and sits there for a very long time.

**Comment by mnsrulz** [comment]
it's such a bad experience..

**Comment by Flywheel** [comment]
I experienced this very annoying spinning cursor on paste issue months ago when using the insiders version.  I switched back to the regular version and forgot about it. But now, out of the blue, it's happening in there as well.
Version: 1.96.2 (user setup)
OS: Windows_NT x64 10.0.26100

**Comment by Mvb1122** [comment]
I also have this bug. What worked for me as a workaround was to turn off copilot suggestions (if you have them on.)

Go under the Github Copilot menu and then select disable completions. 
![Image](https://github.com/user-attachments/assets/2209b1af-9e71-46c3-8336-af010aff9573)

**Comment by superpowered** [comment]
I aslo recently had this issue, and found that I had a new extension "**IntelliCode**" & "**IntelliCode API Usage Examples**" that I don't remember installing. (Potentially added by my company?) Disabling them resolved the issue immediatly

**Comment by thhollev** [comment]
This has also happened to me the last couple days using Version: 1.96.2 (Universal) Commit: fabdb6a30b49f79a7aba0f2ad9df9b399473380f

When trying to copy it makes sense to have some copilot intellisense however it should not block the pasting behaviour.

**Comment by skykep** [comment]
I got tired of fumbling through many websites and threads on this.  Clearly it's a 1.96.2 issue.  I downgraded to 1.95.3 and it's working as normal.

**Comment by mjbvz** [comment]
See https://github.com/microsoft/vscode/issues/235959#issuecomment-2539548324 for a workaround if your project is effected

If you are still seeing this on the current VS Code insiders builds (1.97), please share an example project with steps to reproduce the issue or a log from when this happens

**Comment by luizotcarvalho** [comment]
I disabled this two settings and the loading stoped:

<img width="965" alt="Image" src="https://github.com/user-attachments/assets/3a881d91-81e0-4ee4-b914-6e5c0d301d39" />

**Comment by luke-z** [comment]
Can confirm that my VSCode (1.96.4 and Cursor using VSCode 1.96.2) is also extremely slow. 
Today I waited 1 minute for intellisense to figure out what's going on after adding a variable I previously deleted.

Example:

I have a variable
```const result = '123'```

The variable is used by ```console.log(result)```

When I delete the definition, it takes VSCode about 10 seconds to realize that the variable is gone and then I get the lines below the variable in console.log

It seems like it started this week, last week it felt pretty quick.

**Comment by Jawmo** [comment]
I noticed this recently as well. I thought it was related to my jsconfig/tsconfig file so I excluded some additional folders, which seemed to help but the editor is still quite slow to import items, auto-complete imports, recognize a missing import/varialbe in the file, etc. It also has felt much worse recently.

Is this what you were referring to @luke-z ?

**Comment by luke-z** [comment]
@Jawmo exactly, any change in the code takes a long time to update autocomplete, type issues and intellisense.

If I hover over a new variable it loads 10-20 seconds, sometimes a minute, to display the usual info when hovering over a variable in VSCode

**Comment by mklnln** [comment]
It's a bit of a silly solution, but in a pinch Ctrl+Shift+V can perform the same function as Ctrl+V and I was able to get around this spinning circle problem thanks to that.

**Comment by BRebey** [comment]
Is there a real fix for this yet, short of disabling features as the workarounds above suggest?
